tests/test_other.py

```diff
@@ -2499,6 +2499,7 @@ def test_embind_closure_no_dynamic_execution(self):
   def test_embind(self, extra_args):
     test_cases = [
       (['-lembind']),
+      (['-lembind', '-std=c++17']),
```

Comment:
It looks like `noexcept` was added in C++11 .. so why is this needed? https://en.cppreference.com/w/cpp/language/noexcept_spec

Reply:
The keyword was added, but it didn't become part of the function signature until c++17.  If the changes in bind.h are removed the test will fail to compile under c++17.

Reply:
Maybe worth adding a quick comment on the line above?